src/jdk.compiler/share/classes/com/sun/source/util/DocTrees.java

```diff
@@ -113,27 +139,37 @@ public static DocTrees instance(ProcessingEnvironment env) {
     public abstract DocCommentTree getDocCommentTree(Element e);
 
     /**
-     * Returns the doc comment tree of the given file. The file must be
-     * an HTML file, in which case the doc comment tree represents the
-     * entire contents of the file.
-     * Returns {@code null} if no doc comment was found.
+     * Returns the doc comment tree of the given file, which must
+     * be of one of the supported file types.
+     *
+     * <p>The supported file types are:
+     * <ul>
+     * <li>HTML files, identified by a file name ending in {@code .html},
+     * <li>Markdown files, identified by a file name ending in {@code .md}.
+     * </ul>
      * Future releases may support additional file types.
      *
      * @implNote The default implementation of this method returns a
      * new {@code DocCommentTree} instance for each invocation.
      *
      * @param fileObject the content container
      * @return the doc comment tree
+     * @throws IllegalArgumentException if the file type is not supported
```

Comment:
It's effectively a new exception, since before this work, we simply assumed that any file being read was in HTML format. Now, we check the file type, as described in the modified doc comment.   The entire doc comment is subject to CSR.